service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -509,24 +508,16 @@ private ApiCreateCloudContextResult fetchCreateCloudContextResult(
   }
 
   @Override
-  public ResponseEntity<String> enablePet(UUID workspaceId) {
+  public ResponseEntity<Void> enablePet(UUID workspaceId) {
     AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
     // TODO(PF-1007): This would be a nice use for an authorized workspace ID.
     // Validate that the user is a workspace member, as enablePetServiceAccountImpersonation does
     // not authenticate.
     workspaceService.validateWorkspaceAndAction(
         userRequest, workspaceId, SamConstants.SamWorkspaceAction.READ);
-    String userEmail =
-        SamRethrow.onInterrupted(() -> samService.getUserEmailFromSam(userRequest), "enablePet");
-    String petSaEmail =
-        SamRethrow.onInterrupted(
-            () ->
-                samService.getOrCreatePetSaEmail(
-                    gcpCloudContextService.getRequiredGcpProject(workspaceId), userRequest),
-            "enablePet");
-    UserWithPetSa userAndPet = new UserWithPetSa(userEmail, petSaEmail);
-    petSaService.enablePetServiceAccountImpersonation(workspaceId, userAndPet);
-    return new ResponseEntity<>(petSaEmail, HttpStatus.OK);
+    petSaService.enablePetServiceAccountImpersonation(
+        workspaceId, userRequest.getEmail(), userRequest.getRequiredToken());
```

Comment:
I'm not positive, but I think this email will still need to come from Sam, i.e. keeping the previous
```
 String userEmail =
        SamRethrow.onInterrupted(() -> samService.getUserEmailFromSam(userRequest), "enablePet");
```
The email field of the userRequest gets populated by the proxy in front of WSM, so in places where we run WSM without the proxy (i.e. tests and locally), this might cause issues.

Reply:
Done. I realized this as well, you beat me to it. :) Changed another file as well.